crates/ruff_linter/src/rules/pydocstyle/snapshots/ruff_linter__rules__pydocstyle__tests__D207_D.py.snap

```diff
@@ -1,14 +1,13 @@
 ---
 source: crates/ruff_linter/src/rules/pydocstyle/mod.rs
-snapshot_kind: text
 ---
 D.py:232:1: D207 [*] Docstring is under-indented
     |
 230 |     """Summary.
-231 | 
+231 |
+    | ^ D207
```

Comment:
Hmm, this also seems off but it's possible that it only used to work out of poor luck? I think the idea here is that the violation is at the begining of `Description` because whitespace only lines should be ignored. 

The relevant line seems to be 

https://github.com/astral-sh/ruff/blob/5bc9d6d3aa694ab13f38dd5cf91b713fd3844380/crates/ruff_linter/src/rules/pydocstyle/rules/indent.rs#L226-L227

Adding the `line.start()` seems correct.

Reply:
The offsets reported by the lint here do indeed appear correct (note the assert):

```rust
use annotate_snippets::{Level, Renderer, Snippet};

fn main() {
    let source = "    \"\"\"Summary.\n\nDescription.\n\n    \"\"\"\n";
    assert_eq!(&source[17..], "Description.\n\n    \"\"\"\n");
    let src_annotation = Level::Error.span(17..17).label("D207");
    let snippet =
        Snippet::source(source).line_start(230).annotation(src_annotation);
    let message = Level::Error.title("").snippet(snippet);

    println!("{}", Renderer::plain().render(message));
}
```

But the output of the above is:

```
error
    |
230 |     """Summary.
231 |
    | ^ D207
232 | Description.
233 |
234 |     """
    |
```

So this looks like a legit bug in `annotate-snippets`.

Reply:
Filed upstream: https://github.com/rust-lang/annotate-snippets-rs/issues/177

I'll take a look at fixing this myself.

Reply:
OK, upstream bug was an illusion. Lol. I've fixed this by tweaking the spans produced by the D207 and D208 lints.